### DISCUSSION TITLE
http_request does not exist when used in function trigger

### DISCUSSION THREAD

**Discussion by kvnam** [supabase/supabase#6966] [Questions] [answered]
Hey,

I set up a function trigger to make an api request when Update or Insert ops occur on a table. I enabled HTTP extension, set the function permission to Security Definer (not invoker) and set up the trigger. This works great when I update or insert a record in the table directly through the Supabase UI. Also within the function I'm using `extensions.http_get`, it wouldn't work otherwise.

Now when I trigger an update/insert from my app (using a different Postgres user) I get an error `http_request` does not exist. Any help with what I'm missing here would be great. Thanks!

**Comment by GaryAustin1** [comment, marked as answer]
You likely need the extensions schema in your search path...
Add SET search_path = extensions, public, pg_temp to your function definition

Edit: Or you can add extensions to the search path for the API...
![image](https://user-images.githubusercontent.com/54564956/170067012-47f826ca-8c46-4732-aa8e-160523d0d2cd.png)

**Reply by kvnam** [reply]
Thank you @GaryAustin1 , that did it. Request goes through fine through the app as well now.

**Comment by ultWorld** [comment]
I fixed it with [above answer](https://github.com/orgs/supabase/discussions/6966#discussioncomment-2812677) 

and also had to enable the HTTP extention in Database -> Extensions -> HTTP
![Screenshot from 2023-07-05 02-04-02](https://github.com/supabase/supabase/assets/104983217/d773b82c-60eb-45ab-a485-ea7637405529)

**Comment by shinriyo** [comment]
@ultWorld 
Thank you for your reply.
I also did a similar tutorial. And I was stacked in the error.
https://supabase.com/docs/guides/getting-started/tutorials/with-svelte

I understood what it said "Enable the [http extension for the extensions schema](https://supabase.com/dashboard/project/_/database/extensions) in the Dashboard."
